### PR TITLE
DDF-3261 optimizations for multi-branch pipeline

### PR DESCRIPTION
**Pull request by oconnormi**
#### What does this PR do?
Adds a few optimizations to the multi-branch pipeline support

  * Blocks concurrent builds on the same branch
  * Adds timestamps
  * Incremental owasp for prs
  * Incremental sonar for prs

#### Who is reviewing it? 

@mcalcote @LinkMJB 

#### Select relevant component teams: 

@codice/build 

#### Choose 2 committers to review/merge the PR. 

@shaundmorris 
@clockard 

#### What are the relevant tickets?

[DDF-3261](https://codice.atlassian.net/browse/DDF-3261)

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -138,7 +147,14 @@ pipeline {
...
+                                        script {
+                                            // If this build is not a pull request, run sonar scan. otherwise run incremental scan
+                                            if (env.CHANGE_ID == null) {
+                                                sh 'mvn -q -B -Dfindbugs.skip=true -Dcheckstyle.skip=true org.jacoco:jacoco-maven-plugin:prepare-agent install sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=codice -Dsonar.projectKey=ddf -pl !$DOCS,!$ITESTS'
+                                            } else {
+                                                sh 'mvn -q -B -Dfindbugs.skip=true -Dcheckstyle.skip=true org.jacoco:jacoco-maven-plugin:prepare-agent install sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=codice -Dsonar.projectKey=ddf -pl !$DOCS,!$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
```
Comment (shaundmorris) [inline Jenkinsfile]: ❓ Should this show `preview` mode for the PRs? On the incremental we still don't want to push the results to the sonarqube server that should only be done on the respective branches (master, 2.12.x, etc)

Comment (LinkMJB) [reply Jenkinsfile]: Have we ever tried running a sonar scan on an incremental build? I'm curious what that will look like on the sonarqube site, flipping back and forth between full and incremental build scans.

Comment (oconnormi) [reply Jenkinsfile]: Updated with preview mode, should I remove the incremental build options? Does preview mode push scan results up to the sonar site @shaundmorris? If not than incremental should be fine in preview mode

**Comment by oconnormi** [comment]
I'm voting that we should skip a CI build for this since it won't currently do anything but run the itests on previously deployed artifacts

**Pull request merged**